### DISCUSSION TITLE
Shipping build for android will crash on launch.

### DISCUSSION THREAD

**Discussion by brexGmbH** [maschere/asset-documentation#4] [MicroBenchmark]
Hi,
the plugin causes the app to crash on shipping builds for android.
Debug build works fine.
Maybe a proguard issue?

**Comment by maschere** [comment]
Hi. I will check for issues. Does it also happen in shipping builds for other platforms, i.e. Windows?
And please note that the plugin is supposed to deactivate itself in shipping mode, so that it does not eat unnecessary resources. Though of course it's not supposed to crash.

**Reply by brexGmbH** [reply]
We haven't tested shipping builds on other plattforms so far. We just deactivated the plugin for now.
We also didn't have any blueprints actually using the plugin and it still caused the crash.
Android LogCat showed some error about the "resetAll" function or so, I can't remember exactly.

**Reply by brexGmbH** [reply]
Are you still looking for a solution?

**Reply by maschere** [reply]
I think I found the issue in the ResetAll function like you mentioned (though I cannot test on Android atm). I will submit a fix to Epic, though that usually takes some time. In the meantime, you can fix it in cpp source by changing the body of the ResetAll function in MicroBenchmarkSubsystem.cpp (starting at line 38) to this:

> 	OpenBenchmarks.Empty();
> #if !UE_BUILD_SHIPPING
> 	int32 a = 0;
> 	for (int32 i = 0; i < 100; ++i)
> 	{
> 		a += i;
> 		Tic("Overhead");
> 		Tic("OverheadInner");
> 		Toc("OverheadInner",false);
> 		Toc("Overhead",false);
> 	}
> 	Overhead = BenchmarkResults["Overhead"].AvgNanoSeconds();
> 	OpenBenchmarks.Empty();
> 	BenchmarkResults.Empty();
> 	UE_LOG(LogMicroBenchmarkSubsystem,Display,TEXT("Calibrated overhead of tictoc with %llu nanoseconds"), Overhead.count());
> #endif
